Improved test infrastructure

jhawthorn opened the pull request:
Closely related to #142

I've made an attempt at improving the test setup.

These commits add gemfiles for rails 2.3, 3.0, 3.1, and 3.2. This is somewhat similar to how thoughbot/paperclip is tested. A rails application is initialized in test_helper.rb for rails versions >= 3 which should alleviate the need for a dummy project.

Added a few new rake tasks

```
rake bundles:install # bundle install all available gemfiles
rake bundles:update  # bundle update all available gemfiles
rake bundles:test    # test against all available gemfiles
                     # (this is now the default for `rake`)
```

Tests for one version can be run like

```
BUNDLE_GEMFILE=gemfiles/3.2.gemfile bundle exec rake test
```

Also added a .travis.yml, result can be seen here
http://travis-ci.org/#!/jhawthorn/declarative_authorization/builds/1841883

jhawthorn committed:
require 'test_helper' instead of absolute path

jhawthorn committed:
Adapt test suite for multiple versions of rails

* initialize an app inside test_helper for rails 3
* gemfiles/ contains gemfiles for 2.3, 3.0, 3.1, and 3.2
* rake bundles:test runes tests against all gemfiles
* rake bundles:install and bundles:update installs/updates

jhawthorn committed:
add .travis.yml

stffn commented:
Wow, great patch to set the testing infrastructure right.  One point, though: are you sure that we need to have those *.lock in the repository?

jhawthorn committed:
don't include gemfiles/*.gemfile.lock

jhawthorn committed:
allow travis failures with ruby 1.9 and rails 2.3

Tests do not pass under rails 2.3 because of an issue related to
respond_to_missing? and delegation. see
https://rails.lighthouseapp.com/projects/8994/tickets/5410-multiple-database-queries-when-chaining-named-scopes-with-rails-238-and-ruby-192

jhawthorn commented:
You are right. I've removed *.gemfile.lock.
